projects/graphql-client/src/graphql-apollo-integration.test.ts

```diff
@@ -0,0 +1,243 @@
+
+    // Make second request
+    spectator.service
+      .query({
+        query: gql(buildRequestString()),
+        errorPolicy: 'all',
+        fetchPolicy: 'no-cache'
+      })
+      .subscribe();
+    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
+    spectator.controller.verify();
+    flush();
+  }));
+
+  test('only fires queries that are subscribed', fakeAsync(() => {
+    spectator = createService();
+    const query$ = spectator.service.query({
+      query: gql(buildRequestString()),
+      errorPolicy: 'all',
+      fetchPolicy: 'cache-first'
+    });
+    tick();
+    spectator.controller.expectNone(graphQlUri);
+    query$.subscribe();
+
+    // Flush first request out, otherwise apollo deduping will always reuse an identical in flight query
+    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
+    flush();
+  }));
+
+  // test('does not cache entities, even if they share an id', fakeAsync(() => {
+  //   spectator.service.registerHandler(buildQueryHandler());
+  //   spectator.service.queryImmediately(buildRequest()).subscribe();
+  //   // Return an object to be cached
+  //   spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
+  //   // Make a new query, but tweak params so it's not cached and return same id with differen val
+  //   const modifiedRequest = buildRequest();
+  //   modifiedRequest.select.arguments = [{ name: 'a', value: 'b' }];
+  //   spectator.service.queryImmediately(modifiedRequest).subscribe();
+  //   // Return a response with same id but different value
+  //   const modifiedResponse = buildServerResponse();
+  //   modifiedResponse.data.test.value = 'baz';
+  //   spectator.expectOne(graphQlUri, HttpMethod.POST).flush(modifiedResponse);
+  //   // Now make the first request again - it should be the original result
+  //   spectator.service.queryImmediately(buildRequest()).subscribe(result => {
+  //     expect((result as { value: string }).value).toEqual('bar');
+  //   });
+  //   // Tick, our responses are always async
+  //   tick();
+  // }));
+
+  // test('late subscribers to a debounced query can access results', fakeAsync(() => {
+  //   spectator = createService();
+  //   const query$ = spectator.service.query({
+  //     query: gql(buildRequestString()),
+  //     errorPolicy: 'all',
+  //     fetchPolicy: 'no-cache'
+  //   });
+  //   tick();
+  //   spectator.controller.expectNone(graphQlUri);
+
+  //   tick(10);
+  //   query$.subscribe();
+
+  //   // Flush first request out, otherwise apollo deduping will always reuse an identical in flight query
+  //   expect(spectator.expectOne(graphQlUri, HttpMethod.POST).request.body).toEqual([
+  //     {
+  //       variables: {},
+  //       query: '{\n  test {\n    id\n    value\n    __typename\n  }\n}\n'
+  //     }
+  //   ]);
+  //   flush();
+  // }));
+
+  // test('resubscribing to a query triggers an immediate new request', fakeAsync(() => {
```

Comment:
This test fails now for the apollo observable. Do we really need this?

Reply:
I'd like to get it working but we can drop if it becomes problematic. From looking at the commented out test, I'd suspect the issue is that we're never completing the first request. See from other tests:
```
    // Flush first request out, otherwise apollo deduping will always reuse an identical in flight query
    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
```

Reply:
thanks!